src/searchd.cpp

```diff
@@ -1813,6 +1813,7 @@ void SearchRequestBuilder_c::SendQuery ( const char * sIndexes, ISphOutputBuffer
 			case SPH_FILTER_USERVAR:
 			case SPH_FILTER_STRING:
 				tOut.SendString ( tFilter.m_dStrings.GetLength()==1 ? tFilter.m_dStrings[0].cstr() : nullptr );
+				tOut.SendByte ( (char) tFilter.m_eStrCmpDir );
```

Comment:
you also need to increase search protocol version (SearchdCommandV_e) along with the change of the protocol. However I'd separate the commits and keep the change of master \ agent communication at the end of the development

Reply:
Do you mean I should keep these changes out of the current review cycle, but reintroduce them at the end, in this same PR?

Reply:
yes
it could be better to make separate commit related to distributed index \ master-agent communication to not mix different implementation of different part finished functions

Reply:
Just to clarify, when you say separate commit, do you mean separate pull request altogether?

Reply:
no
just a new commit around the end of this PR, prior to test of all new functions.

Reply:
I've commented those lines out for the purposes of this intermediate PR review cycle.

Reply:
> prior to test of all new functions

Would appreciate some guidance on the way tests are written for Manticore as well

Reply:
you could set up mysql server and php 8 or above then go to `build/test` directory and issue `php uberterst t` and check the output and all failed test `report.txt`
or you could push the code into your own repository and check Actions jobs for example I see failed tests for your [PR#1900](https://github.com/manticoresoftware/manticoresearch/actions/runs/8117713530/job/22190732932?pr=1900) 
```
build/status_debug_test_701_800: failure
```

and looking at the [job](https://github.com/manticoresoftware/manticoresearch/actions/runs/8117713530/job/22190653623?pr=1900) I see the test rt_321 failed
```
 72/102 Test #770: rt_321 -- percolate query ..................................................***Failed    0.58 sec
        Start 770: rt_321 -- percolate query
      Test #770: rt_321 -- percolate query ..................................................***Failed    0.59 sec
        Start 770: rt_321 -- percolate query
      Test #770: rt_321 -- percolate query ..................................................***Failed    0.59 sec
        Start 770: rt_321 -- percolate query
      Test #770: rt_321 -- percolate query ..................................................***Failed    0.58 sec
        Start 770: rt_321 -- percolate query
      Test #770: rt_321 -- percolate query ..................................................***Failed    0.59 sec
```

Reply:
The output of `php ubertest.php t`:

```
==== Run 1 ====
Settings:

SUBTEST FAILED, UNEXPECTED ERROR:
ERROR: source 'srctest': unknown type 'mysql'; skipping.
ERROR: table 'test': failed to configure some of the sources, will not index.
==== Run 2 ====
Settings:

SUBTEST FAILED, UNEXPECTED ERROR:
ERROR: source 'srctest': unknown type 'mysql'; skipping.
ERROR: table 'test': failed to configure some of the sources, will not index.
```

would you happen to know what this issue means?

Reply:
Second question. I'm creating a new test (`/test/test_467`) and I have created `text.xml` to test the functionality I just added. How exactly do I generate a `model.bin`?

Reply:
seems mysql dev package missed that is why you build indexer without MySQL data source, you need to install these dependencies and reconfigure and rebuild indexer https://manual.manticoresearch.com/dev/Installation/Compiling_from_sources#External-dependencies

Or you could use Docker container to build package https://manual.manticoresearch.com/dev/Installation/Compiling_from_sources#Building-using-CI-Docker or follow steps described in the blog https://manticoresearch.com/blog/building-and-testing-manticore-search-manually-with-docker/

Reply:
to generate model you need issue `php ubertest g 467` then check report.txt that results are ok then retest the result model as `php ubertest t 467`